pkg/controller/performanceprofile/components/profile/profile.go

```diff
@@ -177,12 +177,15 @@ func validateNUMA(numa *performancev2.NUMA) error {
 	return nil
 }
 
-func validateCPUCoresGrouping(cpus *performancev2.CPU) error {
-	overlap, err := components.CPUListIntersect(string(*cpus.Reserved), string(*cpus.Isolated))
+func validateCPUCores(cpus *performancev2.CPU) error {
+	cpuLists, err := components.NewCPULists(string(*cpus.Reserved), string(*cpus.Isolated))
 	if err != nil {
-		return validationError(fmt.Sprintf("internal error: %v", err))
+		return validationError(fmt.Sprintf("internal error parsing cpus: %v", err))
 	}
-	if len(overlap) != 0 {
+	if numIsolated := cpuLists.CountIsolated(); numIsolated == 0 {
```

Comment:
nit: you do not need here a new variable
```
if cpuLists.CountIsolated() == 0
```
should work as well, but if from your perspective it has better readability I will not resist it :)

Reply:
I like your way more!